mantle/platform/platform.go

```diff
@@ -500,6 +500,17 @@ func CheckMachine(ctx context.Context, m Machine) error {
 		if err != nil {
 			return fmt.Errorf("failed to query systemd for failed units: %s: %v: %s", out, err, stderr)
 		}
+		// Ensure sssd is not stuck in activating state
+		// https://github.com/coreos/coreos-assembler/issues/2798
+		// https://bugzilla.redhat.com/show_bug.cgi?id=2072050		
+		var NRestarts int
+		stuckUnits := "sssd"
```

Comment:
I don't think we want to check only a single service, but rather all the services on the host.

We should be able to query the system for units in `activating` and then check the `NRestarts` value.  For example:

```bash
[core@cosa-devsh ~]$ systemctl --no-legend --state failed list-units
[core@cosa-devsh ~]$ systemctl --no-legend --state activating list-units
sssd.service loaded activating start start System Security Services Daemon
$ systemctl show -p NRestarts --value sssd
16
```

Since it may take some time for a service to start and it is possible the service was restarted in the process of starting (i.e. `systemd-journald`, see below), we should set a limit on the amount of restarts that we trigger on. 

```bash
$ systemctl show -p NRestarts systemd-journald
NRestarts=1
$ systemctl show -p NRestarts sssd            
NRestarts=2
```

That output was manually gathered after booting RHCOS; so I would say we can safely expect a service to restart 1 time before being successful.  Anything more than than is problematic?

Reply:
Thanks @miabbott for your detail correction. I will update